### DISCUSSION TITLE
Non-cuboid portals

### DISCUSSION THREAD

**Discussion by ColaIan** [Zrips/CMI#2066] [Feature Requests]
Allow the creation of non-cuboid portals by configuring portal blocks instead of a 2 point regions.

---
### Detailed explanation of what it should do, when and in what situation: ###
Currently, portals are created with a 2 point cuboid regions, and I hope we could place some "placeholder blocks", then right click, then the placeholder blocks get removed and are set as the portal blocks

### Basic explanation as to why this feature is needed and/or useful to have: ###
this is useful for creating irregular portals like circle, star, pentagon, triangle, and etc

**Comment by CloudeLecaw** [comment]
Well yes it would be nice to be able to also creat Non Cubiod Portals <3
![image](https://user-images.githubusercontent.com/12080017/61440436-ca183d80-a943-11e9-86f2-91c554048154.png)

**Comment by ColaIan** [comment]
This is still rectangular though
I can technically do this: 
![image](https://user-images.githubusercontent.com/43201383/61441158-99ea9200-a977-11e9-9c9d-c3302b6a03b4.png)

But not this:
![image](https://user-images.githubusercontent.com/43201383/61441193-ad95f880-a977-11e9-99bd-fb4ce9d1e133.png)

**Comment by CloudeLecaw** [comment]
So this is a single CMI Portal without just using just two points like the cuiboid once? (mark point1, point2 and have a square...)

i know the target would be to be able to create Portals like this : - D
![image](https://user-images.githubusercontent.com/12080017/61459489-1c218900-a96d-11e9-8d9f-eb1b04d0ded3.png)

**Comment by Zrips** [comment]
This could be some what resolved by adding option to define multiple cuboids for same portal, which would allow to select multiple areas which would create none cuboid one. even tho it would be made of bunch cuboids. In general, cuboid is simplest and most efficient way in checking if player is in area, having  complex area will involve complex calculations which will only increase load on server. So its a trade off and entire code is made to deal with cuboids at this stage, so changing it to support something more complex would require quite allot of recode.
